wrpclient/mocks_test.go

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2022 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wrpclient
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+type mockHTTPClientSuccess struct{}
```

Comment:
I would turn all of this into a single `mockHTTPClient` struct that embeds `mock.Mock`.  Then the mock can expected to be able to Get an int and a `[]byte`.  Calling the mock would look something like:
```golang
var payload []byte
err := wrp.NewEncoderBytes(&payload, wrp.JSON).Encode(&wrp.Message{Type: wrp. SimpleRequestResponseMessageType}
require.NoError(t, err)
m := new(mockHTTPClient)
m.On("Do", mock.Anything).Return(http.StatusOK, payload)
```
One byproduct of this is that the caller of the mock would be responsible for encoding the wrp.  Since a few use cases use the same encoding anyways, this will allow us to encode the same wrp just once instead of multiple times.


---
In my example code, I switched out ints for their enum values.  This will make for much more human readable code.